### PR TITLE
Use Stage 1 For Non Finalized Sync

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Sync Improvement

**What does this PR do? Why is it needed?**

- [x] Allows the 2nd stage of sync to re-use the block queue from stage 1, therefore allowing it the ability to crunch through
large periods of skipped slots and non-finality.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/blockchain/chain_info.go [hunk 1]
```diff
@@ -189,6 +190,14 @@ func (s *Service) HeadETH1Data() *ethpb.Eth1Data {
 	return s.head.state.Eth1Data()
 }
 
+// HeadFinalizedEpoch returns the finalized epoch of the current head state.
+func (s *Service) HeadFinalizedEpoch() uint64 {
```
Comment (terencechain) [inline beacon-chain/blockchain/chain_info.go:194]: I think we should be using the latest cached finalized epoch. Not the finalized epoch from the head. Keep in mind, the finalized epoch from head may go down but latest cached finalized epoch can never go down

File: beacon-chain/sync/initial-sync/round_robin.go [hunk 2]
```diff
@@ -189,6 +139,29 @@ func (s *Service) processFetchedData(
...
 
+// processFetchedData processes data received from queue.
+func (s *Service) processFetchedDataRegSync(
+	ctx context.Context, genesis time.Time, startSlot uint64, data *blocksQueueFetchedData) {
+	defer func() {
+		if !featureconfig.Get().EnablePeerScorer || data.pid == "" {
+			return
+		}
+		scorer := s.p2p.Peers().Scorers().BlockProviderScorer()
+		if diff := s.chain.HeadSlot() - startSlot; diff > 0 {
```
Comment (prestonvanloon) [inline beacon-chain/sync/initial-sync/round_robin.go:150]: Could this underflow? Maybe change to "if diff is not zero" 

```suggestion
		if diff := s.chain.HeadSlot() - startSlot; diff != 0 {
```

Comment (prestonvanloon) [reply beacon-chain/sync/initial-sync/round_robin.go:150]: Actually, what about `s.chain.HeadSlot() != startSlot`?

```suggestion
		if s.chain.HeadSlot() != startSlot {
```

Comment (farazdagi) [reply beacon-chain/sync/initial-sync/round_robin.go:150]: Will fix both in a moment.

**Pull request merged**